macos/generate_wazuh_packages.sh

```diff
@@ -119,7 +126,7 @@ function sign_pkg() {
 function build_package() {
 
     # Download source code
-    git clone --depth=1 -b ${BRANCH_TAG} ${WAZUH_SOURCE_REPOSITORY} "${WAZUH_PATH}"
+    git clone --depth=1 -b ${BRANCH_TAG} ${WAZUH_SOURCE_REPOSITORY} "${WAZUH_PATH}" || clean_and_exit 1
```

Comment:
`clean_and_exit` should be executed outside.

Reply:
Done in https://github.com/wazuh/wazuh-packages/pull/329/commits/b52cef11212f5b467eeffe8cde9257dcc8a4dfb0.